modules/docs/lib/widgets/enhancedComponent.tsx

```diff
@@ -0,0 +1,199 @@
+/**
+ * Used to keep track of JSDoc in subcomponent definitions.
+ *
+ * For example:
+ * ```tsx
+ * subComponents: {
+ *   // JSDoc goes here instead of on `TabsItem` for better dev experience
+ *   Item: TabsItem
+ * }
+ * ```
+ */
+const ParentComponentJSDocContext = React.createContext(defaultJSDoc);
+
+const fileNameToCategoryMap: Record<string, string[]> = {
+  Layout: [
+    'utils/layout',
+    'utils/gridArea',
+    'utils/gridItem',
+    'utils/flexItem',
+    'utils/position',
+    'utils/space',
+    'utils/stack',
+  ],
+  Color: ['utils/color', 'utils/background'],
+  Border: ['utils/border'],
+  Depth: ['utils/depth'],
+  Text: ['utils/text'],
+  Other: ['utils/other'],
+};
+
+registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta}) => {
+  const groups = groupProps(value.props);
+  const parentComponentName = React.useContext(ParentComponentNameContext);
+  const parentComponentJSDoc = React.useContext(ParentComponentJSDocContext);
+  const hideHeader = meta && meta.hideHeader === true;
+
+  const intro = (
+    <>
+      {!parentComponentName && !hideHeader && value.displayName ? (
+        <Heading>{value.displayName}</Heading>
+      ) : null}
+
+      <ParentComponentNameContext.Provider value="">
+        <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+          <MdxJSToJSX>{parentComponentJSDoc.description || doc?.description || ''}</MdxJSToJSX>
+        </ParentComponentJSDocContext.Provider>
+      </ParentComponentNameContext.Provider>
+    </>
+  );
+
+  // We don't want to re-document a subcomponent that is a container component with a different
+  // model. If we detect another container, only render the intro and a link to the container
+  // component
+  if (parentComponentName && value.componentType === 'container') {
+    return (
+      <>
+        {intro}
+        <MDX as="p">
+          This component references the{' '}
+          <ParentComponentNameContext.Provider value="">
+            <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+              <SymbolDialog value={{kind: 'symbol', name: value.displayName || ''}} />
+            </ParentComponentJSDocContext.Provider>
+          </ParentComponentNameContext.Provider>{' '}
+          component.
+        </MDX>
+      </>
+    );
+  }
+
+  return (
+    <>
+      {intro}
+      {value.styleComponent ? (
+        <>
+          <Heading headingOffset={1}>Style Component</Heading>
+          <MDX as="p">
+            <code>
+              <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+                <Value value={value.styleComponent} />
+              </ParentComponentJSDocContext.Provider>
+            </code>
+          </MDX>
+        </>
+      ) : null}
+      <Heading headingOffset={1}>Props</Heading>
+      {value.baseElement && (
+        <MDX as="p">
+          Props extend from <Value value={value.baseElement} />. Changing the <code>as</code> prop
```

Comment:
The contents of the popup triggered by `<Value>` seem off to me if the props extend from another component. For example, this is what we see for ActionBarBar.Item whose props extend from SecondaryButton:

<img width="1051" alt="image" src="https://user-images.githubusercontent.com/146020/221324367-7b337bee-269e-4a6b-bea5-7151c876f746.png">

The content in the popup for SecondaryButton makes sense to me, but I wasn't expecting any content related to ActionBar.Item in the popup.

Reply:
Ah. Good catch. There is a React Context that sets the JSDoc context for subcomponents, but that is not being reset when it comes to links.